dotCMS/src/integration-test/java/com/dotmarketing/filters/FiltersTest.java

```diff
@@ -163,120 +164,90 @@ public String answer(InvocationOnMock invocation) throws Throwable {
-            Logger.info(this.getClass(),
-                    "/testLink1 should forward to /about-us/" + CMSFilter.CMS_INDEX_PAGE);
-            HttpServletRequest request = getMockRequest(site.getHostname(), "/testLink1");
-            filter.doFilter(request, response, chain);
-            Assert.assertEquals(200, response.getStatus());
-            Assert.assertEquals("/about-us/" + CMSFilter.CMS_INDEX_PAGE,
-                    request.getAttribute(Constants.CMS_FILTER_URI_OVERRIDE));
-
-            Logger.info(this.getClass(),
-                    "/testLink2 should forward to /about-us/" + CMSFilter.CMS_INDEX_PAGE);
-            request = getMockRequest(site.getHostname(), "/testLink2");
-            response = new MockResponseWrapper(Mockito.mock(HttpServletResponse.class));
-            filter.doFilter(request, response, chain);
-            Assert.assertEquals(200, response.getStatus());
-            Assert.assertEquals("/about-us/" + CMSFilter.CMS_INDEX_PAGE,
-                    request.getAttribute(Constants.CMS_FILTER_URI_OVERRIDE));
+        vanityUrl1 = filtersUtil.createVanityUrl("test link1", Host.SYSTEM_HOST, "/" + uniqueUrl + "1",
+                "/about-us/" + CMSFilter.CMS_INDEX_PAGE, 200, 1, defaultLanguageId);
+        filtersUtil.publishVanityUrl(vanityUrl1);
 
-            Logger.info(this.getClass(),
-                    "/testLink3 should redirect to http://demo.dotcms.com/about-us/"
-                            + CMSFilter.CMS_INDEX_PAGE);
-            request = getMockRequest(site.getHostname(), "/testLink3");
-            response = new MockResponseWrapper(Mockito.mock(HttpServletResponse.class));
-            filter.doFilter(request, response, chain);
-            Assert.assertEquals(301, response.getStatus());
-            Assert.assertEquals("http://demo.dotcms.com/about-us/" + CMSFilter.CMS_INDEX_PAGE,
-                    response.getRedirectLocation());
+        vanityUrl2 = filtersUtil.createVanityUrl("test link2", site.getIdentifier(),
+                "/" + uniqueUrl + "2", "/about-us/" + CMSFilter.CMS_INDEX_PAGE, 200, 1,
+                defaultLanguageId);
+        filtersUtil.publishVanityUrl(vanityUrl2);
 
-            Logger.info(this.getClass(),
-                    "/testLink4 should redirect to http://demo.dotcms.com/about-us/"
-                            + CMSFilter.CMS_INDEX_PAGE);
-            request = getMockRequest(site.getHostname(), "/testLink4");
-            response = new MockResponseWrapper(Mockito.mock(HttpServletResponse.class));
-            filter.doFilter(request, response, chain);
-            Assert.assertEquals(301, response.getStatus());
-            Assert.assertEquals("http://demo.dotcms.com/about-us/" + CMSFilter.CMS_INDEX_PAGE,
-                    response.getRedirectLocation());
+        vanityUrl3 = filtersUtil.createVanityUrl("test link3", site.getIdentifier(),
+                "/" + uniqueUrl + "3", "http://demo.dotcms.com/about-us/" + CMSFilter.CMS_INDEX_PAGE,
```

Comment:
Codacy found an issue: [The String literal "http://demo.dotcms.com/about-us/" appears 4 times in this file; the first occurrence is on line 188](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)